fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapper.kt

```diff
@@ -13,23 +18,100 @@ import org.mozilla.fenix.settings.SupportUtils
  * Returns a list of all the required Nimbus 'cards' that have been converted to [OnboardingPageUiData].
  */
 internal fun Collection<OnboardingCardData>.toPageUiData(
+    context: Context,
     showNotificationPage: Boolean,
     showAddWidgetPage: Boolean,
-): List<OnboardingPageUiData> =
-    filter {
-        when (it.cardType) {
-            OnboardingCardType.NOTIFICATION_PERMISSION -> {
-                it.enabled && showNotificationPage
-            }
-            OnboardingCardType.ADD_SEARCH_WIDGET -> {
-                it.enabled && showAddWidgetPage
-            }
-            else -> {
-                it.enabled
+): List<OnboardingPageUiData> {
+    val junoOnboardingFeature = FxNimbus.features.junoOnboarding.value()
+    val jexlConditions = junoOnboardingFeature.conditions
+
+    // we are first filtering the cards based on Nimbus configuration
+    return filter { it.shouldDisplayCard(context, jexlConditions) }
+        // we are then filtering again based on device capabilities
+        .filter {
+            when (it.cardType) {
+                OnboardingCardType.NOTIFICATION_PERMISSION -> {
+                    it.enabled && showNotificationPage
+                }
+
+                OnboardingCardType.ADD_SEARCH_WIDGET -> {
+                    it.enabled && showAddWidgetPage
+                }
+
+                else -> {
+                    it.enabled
+                }
```

Comment:
💡 Whenever lambda is more than one line, consider extracting into a function, that would give a name to the logic and increase readability implicitly. Also the filter above is following the same pattern.